create-image

```diff
@@ -0,0 +1,422 @@
+		echo "$0: Warning: no gpg key found in \
+		    keys/$distro_l-$release" 1>&2
+	fi
+
+}
+
+iso_file=
+iso_sha=
+get_iso_file_sha() {
+	local file
+
+	if [[ -n $iso_file ]]; then
+		return
+	fi
+
+	echo "Checking to see if we have the iso for $distro $release:"
+	for file in sha256sum.txt sha256sum.txt.asc; do
+		curl -s -o $iso_dir/$file $ISO_URL/$file
+	done
+	gpg --verify $iso_dir/sha256sum.txt.asc
+
+	# As new dot releases come out, the name of the iso file changes
+	# according to a predictable pattern.  The sha256sums.txt file contains
+	# the proper name.
+	eval $(awk -v pattern="$DVD_PATTERN" \
+	    '$2 ~ pattern { printf("iso_sha=%s; iso_file=%s;", $1, $2) }' \
+	    $iso_dir/sha256sum.txt)
+	if [[ -z $iso_file || -z $iso_sha ]]; then
+		echo "$0: unable to determine ISO file and/or sha256" 1>&2
+		exit 1
+	fi
+}
+
+fetch_iso() {
+	local file sha out
+	local need_iso=false
+	local found_key=false
+
+	if [[ ! -d $iso_dir ]]; then
+		mkdir -p $iso_dir
+	fi
+
+	import_gpg_keys
+	get_iso_file_sha
+
+	echo "Checking integrity of $iso_file"
+	if [[ ! -f $iso_dir/$iso_file ]]; then
+		echo "No local copy of $iso_file. Fetching latest $iso_file"
+		need_iso=true
+	elif ! echo $iso_sha $iso_dir/$iso_file | sha256sum -c -; then
+		echo "Local copy of $iso_file is bad." \
+		    "Fetching latest $iso_file."
+		rm -f "$iso_dir/$iso_file"
+		need_iso=true
+	fi
+
+	if $need_iso; then
+		curl -o "$iso_dir/$iso_file" "$ISO_URL/$iso_file"
+		echo "Checking integrity of $iso_dir/$iso_file"
+		echo "$iso_sha" "$iso_dir/$iso_file" | sha256sum -c -
+	fi
+}
+
+mount_iso() {
+	local file=$1
+	local mntpt=$2
+
+	# Check if $mntpt is already mounted
+	# This may happen if a previous build failed
+	if df -h "$mntpt" 2>/dev/null | grep -qw "$mntpt"; then
+		echo "Unmounting $mntpt from previous build..."
+		umount "$mntpt"
+	fi
+
+	echo "Mounting $file to $mntpt"
+	if [[ ! -d $mntpt ]]; then
+		echo "Creating $mntpt..."
+		mkdir -p "$mntpt"
+	fi
+	case $(uname -s) in
+		Linux)
+			mount "$file" "$mntpt" -o loop
+			;;
+		SunOS)
+			mount -F hsfs "$file" "$mntpt"
+			;;
+		*)
+			echo "$0: do not know how to mount an iso on" \
+			    "$(uname -s)". 1>&2
+			exit 2
+			;;
+	esac
+}
+
+create_ks_root() {
+	local src_mntpt=/tmp/mnt/$distrel
+
+	echo "Creating ISO Layout"
+
+	if [[ -d $ks_root ]]; then
+		echo "Layout $ks_root exists...nuking"
+		rm -rf "$ks_root"
+	fi
+	echo "Creating $ks_root"
+	mkdir -p "$ks_root"
+
+	if [[ -d $custom_rpms ]]; then
+		echo "Copying custom RPMS"
+		find "$custom_rpms" -type f -exec cp {} "$ks_root/Packages" \;
+	fi
+
+	if [[ -f $custom_rpms.remote ]]; then
+		echo "Downloading custom RPMS"
+		local rpm
+		mkdir -p "$ks_root/Packages"
+		pushd "$ks_root/Packages" >/dev/null
+		for rpm in $(sed 's/#.*//' $custom_rpms.remote); do
+			echo "  $(basename $rpm)"
+			curl -O "$rpm"
+		done
+		popd >/dev/null
+	fi
+
+	echo "Finished Populating Layout"
+}
+
+copy_ks_cfg() {
+	echo "Copying Kickstart file"
+	cp $ks_cfg $ks_root/ks.cfg
```

Comment:
Oh damn, I think I understand why you can't factor things out.

Reply:
Right.  Making one commmon ks.cfg is pretty much impossible because anaconda (the program that uses all of this) and the OS as a whole has different features/packages over time.  There's surely a way to add a bunch of standard files and tools to the various install media, but I'm not sure the payback is there to spend a lot of time on it.